src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -143,24 +154,47 @@ internal async Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount
             }
         }
 
+        internal async Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount, IEnumerable<string> languages)
+        {
+            foreach (string language in languages)
+            {
+                var rpcWorkerChannel = _rpcWorkerChannelFactory.Create(_scriptOptions.RootScriptPath, language, _metricsLogger, attemptCount, _workerConfigs);
+                _jobHostLanguageWorkerChannelManager.AddChannel(rpcWorkerChannel, language);
+                await rpcWorkerChannel.StartWorkerProcessAsync();
+                _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", language, rpcWorkerChannel.Id);
+
+                // if the worker is indexing, we will not have function metadata yet so we cannot perform these next three lines
+                if (!_workerIndexing)
+                {
+                    rpcWorkerChannel.SetupFunctionInvocationBuffers(_functions);
+                    rpcWorkerChannel.SendFunctionLoadRequests(_managedDependencyOptions.Value, _scriptOptions.FunctionTimeout);
+                }
+            }
+            SetFunctionDispatcherStateToInitializedAndLog();
+        }
+
         private void SetFunctionDispatcherStateToInitializedAndLog()
         {
             State = FunctionInvocationDispatcherState.Initialized;
             // Do not change this log message. Vs Code relies on this to figure out when to attach debugger to the worker process.
             _logger.LogInformation("Worker process started and initialized.");
         }
 
-        internal async Task InitializeWebhostLanguageWorkerChannel()
+        internal async Task InitializeWebhostLanguageWorkerChannel(IEnumerable<string> languages = null)
         {
-            _logger.LogDebug("Creating new webhost language worker channel for runtime:{workerRuntime}.", _workerRuntime);
-            IRpcWorkerChannel workerChannel = await _webHostLanguageWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
-
-            // if the worker is indexing, we will not have function metadata yet so we cannot perform the next two lines
-            if (!_workerIndexing)
+            languages ??= new[] { _workerRuntime };
+            foreach (string language in languages)
```

Comment:
Not necessarily, we since we have added multiple language support, they can be multiple, but since languages has a default and is not provided we fallback to the default scenario of starting `_workerRuntime`.